packages/devtools_app/test/utils_test.dart

```diff
@@ -911,6 +911,57 @@ void main() {
         expect(['a', 'b'].joinWith('z'), equals(['a', 'z', 'b']));
       });
     });
+
+    group('ListValueNotifier', () {
+      ListValueNotifier<int> notifier;
+
+      bool didNotify;
+
+      setUp(() {
+        didNotify = false;
+        notifier = ListValueNotifier<int>([]);
+        notifier.addListener(() {
+          didNotify = true;
+        });
+      });
+
+      test('notifies on value change', () {
+        expect(didNotify, isFalse);
+        expect(notifier.value, equals([]));
+        notifier.value = [1, 2, 3];
+        expect(didNotify, isTrue);
+        expect(notifier.value, equals([1, 2, 3]));
+      });
+
+      test('notifies on addAndNotify', () {
+        expect(didNotify, isFalse);
+        expect(notifier.value, equals([]));
+        notifier.add(1);
+        expect(didNotify, isTrue);
+        expect(notifier.value, equals([1]));
+      });
+
+      test('notifies on addAllAndNotify', () {
+        expect(didNotify, isFalse);
+        expect(notifier.value, equals([]));
+        notifier.addAll([1, 2]);
+        expect(didNotify, isTrue);
+        expect(notifier.value, equals([1, 2]));
+      });
+
+      test('notifies on forceNotify', () {
+        expect(didNotify, isFalse);
+        expect(notifier.value, equals([]));
+        notifier.value = [1, 2];
+        expect(didNotify, isTrue);
+        expect(notifier.value, equals([1, 2]));
+
+        didNotify = false;
+        notifier.forceNotify();
+        expect(didNotify, isTrue);
+        expect(notifier.value, equals([1, 2]));
+      });
+    });
```

Comment:
Maybe a clear if added to base class and remove/forceNotify?

Reply:
added clear. forceNotify is still useful for the case where a list's contents were updated and add or addAll could not be used